### DISCUSSION TITLE
Problem with enabling DATABASE_URL

### DISCUSSION THREAD

**Discussion by Savke** [dani-garcia/vaultwarden#1957] [Q&A] [answered]
Hi,
I have problem when switching from SQL lite to Postgres or MySql (same problem). I got error like this 

`Error loading config:
	DATABASE_URL is a MySQL URL, but the 'mysql' feature is not enabled`
	
Can anyone help me. tnx

P.S. I don't use docker

**Comment by BlackDex** [comment]
Then you need to compile it with all the features enabled.
`cargo build --features sqlite,mysql,postgresql`

**Reply by Savke** [reply]
I try it with fresh installation with : `cargo build --features postgresql ` and got this error message:
 Compiling reqwest v0.11.4
   Compiling migrations_macros v1.4.2
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-m64" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.0.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.1.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.10.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.11.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.12.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.13.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.14.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.15.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.2.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.3.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.4.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.5.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.6.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.7.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.8.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.migrations_macros.695e1844-cgu.9.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.3ricgereplwtepmv.rcgu.o" "/root/vaultwarden/target/release/deps/migrations_macros-f3e4e8aff0149378.4dlttm48fiq1mva0.rcgu.o" "-Wl,--as-needed" "-L" "/root/vaultwarden/target/release/deps" "-L" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/root/vaultwarden/target/release/deps/libsyn-439e609de190c91e.rlib" "/root/vaultwarden/target/release/deps/libquote-7fb4205e178203ee.rlib" "/root/vaultwarden/target/release/deps/libproc_macro2-f761a0cbcba2cbb6.rlib" "/root/vaultwarden/target/release/deps/libunicode_xid-39ed68676ed2a432.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libproc_macro-3062b5aca2230d0a.rlib" "/root/vaultwarden/target/release/deps/libmigrations_internals-447b0d58e6600e8f.rlib" "/root/vaultwarden/target/release/deps/libdiesel-75e1ab0026cf1e7b.rlib" "/root/vaultwarden/target/release/deps/libpq_sys-1a92f7e41fdf660a.rlib" "/root/vaultwarden/target/release/deps/libchrono-a6a09fddeb25ad83.rlib" "/root/vaultwarden/target/release/deps/libserde-44165420ac41f967.rlib" "/root/vaultwarden/target/release/deps/libnum_integer-d99071f439bd7bef.rlib" "/root/vaultwarden/target/release/deps/libnum_traits-20b2198091f66067.rlib" "/root/vaultwarden/target/release/deps/libtime-d07bd5b9044c9134.rlib" "/root/vaultwarden/target/release/deps/libr2d2-fc76912ce56527f7.rlib" "/root/vaultwarden/target/release/deps/libscheduled_thread_pool-6b93ef3a3fe671b2.rlib" "/root/vaultwarden/target/release/deps/libparking_lot-f29d68e4a4952e3d.rlib" "/root/vaultwarden/target/release/deps/libparking_lot_core-5940d05b804be511.rlib" "/root/vaultwarden/target/release/deps/liblibc-f30ab95b74620eef.rlib" "/root/vaultwarden/target/release/deps/libsmallvec-c47059b310fca87d.rlib" "/root/vaultwarden/target/release/deps/liblock_api-7dcb352b4002c14b.rlib" "/root/vaultwarden/target/release/deps/libscopeguard-dd4d75e185d854bb.rlib" "/root/vaultwarden/target/release/deps/libinstant-421b94d1bcefb917.rlib" "/root/vaultwarden/target/release/deps/liblog-44adb7bb05b080fe.rlib" "/root/vaultwarden/target/release/deps/libcfg_if-017df4c2d4b73aea.rlib" "/root/vaultwarden/target/release/deps/libbyteorder-6f15e771d2bf1a06.rlib" "/root/vaultwarden/target/release/deps/libbitflags-e09810bcc8306198.rlib" "-Wl,--start-group" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-1dd6c19f0820d51d.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-4695c0f6311791db.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libminiz_oxide-1a282f8b292d9e3f.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libadler-a54ae5159230894d.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libobject-13671d003abcccd5.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libmemchr-7509adad918d4b3b.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libaddr2line-66d0be1f7efaf5ca.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgimli-41443b46a1557b2c.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd_detect-60812e99072283ff.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-943ba0c1e3f87a89.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libhashbrown-0dbc7e011696d844.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-78b2343cc72ff57a.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-65fa19680bfb5ee4.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-72cab8079f9b3b1e.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-32d13a5363efe47b.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-9ee1d5d15e6abbeb.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-52d5241975807511.rlib" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-9924c22ae1efcf66.rlib" "-Wl,--end-group" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-96219fb718f2f3e8.rlib" "-Wl,-Bdynamic" "-lpq" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-znoexecstack" "-L" "/root/.rustup/toolchains/nightly-2021-08-22-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/root/vaultwarden/target/release/deps/libmigrations_macros-f3e4e8aff0149378.so" "-Wl,--gc-sections" "-shared" "-Wl,-zrelro" "-Wl,-znow" "-nodefaultlibs"
  = note: /usr/bin/ld: cannot find -lpq
          collect2: error: ld returned 1 exit status


error: could not compile `migrations_macros` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed

**Comment by BlackDex** [comment, marked as answer]
Please read: https://github.com/dani-garcia/vaultwarden/wiki/Building-binary

**Reply by Savke** [reply]
That's it. Tnx man
